### PR TITLE
Test for invalid instructions

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/unittests/parser_test.cpp [hunk 1]
```diff
@@ -797,6 +797,116 @@ TEST(parser, code_section_with_memory_grow)
...
+    const auto section_contents = make_vec({code_bin});
+    const auto bin = bytes{wasm_prefix} + make_section(10, section_contents);
+
+    EXPECT_THROW_MESSAGE(parse(bin), parser_error, "invalid instruction 68");
+}
+
+TEST(parser, code_section_invalid_instruction)
+{
+    const uint8_t invalid_instructions[] = {
+        0x06,
```
Comment (axic) [inline test/unittests/parser_test.cpp]: clang-format translated it into this long list.

What is preferred:
1) This long list as is
2) This long list formatted differently
3) Using vector/array/etc. and pushing the ranges to it

File: test/unittests/parser_test.cpp [hunk 2]
```diff
@@ -797,6 +797,116 @@ TEST(parser, code_section_with_memory_grow)
     EXPECT_THROW_MESSAGE(parse(bin_invalid), parser_error, "invalid memory index encountered");
 }
 
+TEST(parser, code_section_unsupported_f32_const)
+{
+    const auto func_bin =
+        "00"  // vec(locals)
+        "440b"_bytes;
```
Comment (axic) [inline test/unittests/parser_test.cpp]: Will also extend this test to contain every FP instruction.

File: test/unittests/parser_test.cpp [hunk 3]
```diff
@@ -797,6 +797,116 @@ TEST(parser, code_section_with_memory_grow)
...
+    for (const auto instr : invalid_instructions)
+    {
+        const auto func_bin = "00"_bytes  // vec(locals)
+                              + bytes{instr};
+        const auto code_bin = add_size_prefix(func_bin);
+        const auto section_contents = make_vec({code_bin});
+        const auto bin = bytes{wasm_prefix} + make_section(10, section_contents);
+
+        const auto expected_msg = std::string{"invalid instruction "} + std::to_string(instr);
+        EXPECT_THROW_MESSAGE(parse(bin), parser_error, expected_msg.c_str());
```
Comment (axic) [inline test/unittests/parser_test.cpp]: For some reason `EXPECT_THROW_MESSAGE(parse(bin), parser_error, std::string{"invalid instruction "} + std::to_string(instr))` was not compiling.

File: test/unittests/parser_test.cpp [hunk 4]
```diff
@@ -797,6 +797,116 @@ TEST(parser, code_section_with_memory_grow)
...
+        0xf6,
+        0xf7,
+        0xf8,
+        0xf9,
+        0xfa,
+        0xfb,
+        0xfc,
+        0xfd,
+        0xfe,
+        0xff,
```
Comment (chfast) [inline test/unittests/parser_test.cpp]:
```suggestion
        0xff
```
Remove tailing `,` to have it compacted.

**Comment by axic** [comment]
@chfast how about this then?

**Comment by chfast** [comment]
> @chfast how about this then?

Ok.

**Pull request merged**